plan/rule_partition_processor.go

```diff
@@ -105,6 +105,12 @@ func (s *partitionProcessor) prune(ds *DataSource) (LogicalPlan, error) {
 		newDataSource.baseLogicalPlan = newBaseLogicalPlan(ds.context(), TypeTableScan, &newDataSource)
 		newDataSource.isPartition = true
 		newDataSource.partitionID = pi.Definitions[i].ID
+		// Find a better way to handle here.
+		// There are so many place that refer to the old DataSource including the schema
```

Comment:
The comment is confusion, I think you should directly explain why the `newDataSource.id` is set to the `ds.id`. It's not because we hold the reference of the `Datasource`, it's because we use the `id` as the `FromID` of the `Column`s in the `Expression` of parent.

Reply:
"use the id as the FromID of the Columns in the Expression of parent"
This is a kind of reference!

Reply:
Yes, I mean, we should statement the comment in a straightforward way.